table/index_builder.h

```diff
@@ -145,6 +145,8 @@ class ShortenedIndexBuilder : public IndexBuilder {
     return index_block_builder_.CurrentSizeEstimate();
   }
 
+  friend class PartitionedIndexBuilder;
```

Comment:
To access its index_block_builder_ when creating the flush policy: https://github.com/facebook/rocksdb/pull/2041/files/7a6479a8894a51945bd36b59ef829ccbec50ec22#diff-cbb7d5d97c7da536b506a38dd250f053R75